prog/minio/minio_server_nexus.rb

```diff
@@ -154,12 +174,30 @@ def available?
     client = Minio::Client.new(
       endpoint: minio_server.connection_string,
       access_key: minio_server.cluster.admin_user,
-      secret_key: minio_server.cluster.admin_password
+      secret_key: minio_server.cluster.admin_password,
+      ssl_ca_file_data: minio_server.cluster.root_certs
     )
 
     server_data = JSON.parse(client.admin_info.body)["servers"].find { _1["endpoint"] == minio_server.endpoint }
     server_data["state"] == "online" && server_data["drives"].all? { _1["state"] == "ok" }
   rescue
     false
   end
+
+  def create_certificate
+    root_cert = OpenSSL::X509::Certificate.new(minio_server.cluster.root_cert_1)
+    root_cert_key = OpenSSL::PKey::EC.new(minio_server.cluster.root_cert_key_1)
+    if root_cert.not_after < Time.now + 60 * 60 * 24 * 365 * 1
+      root_cert = OpenSSL::X509::Certificate.new(minio_server.cluster.root_cert_2)
+      root_cert_key = OpenSSL::PKey::EC.new(minio_server.cluster.root_cert_key_2)
+    end
+
+    Util.create_certificate(
+      subject: "/C=US/O=Ubicloud/CN=#{minio_server.cluster.ubid} Server Certificate",
+      extensions: ["subjectAltName=DNS:#{minio_server.cluster.hostname},DNS:#{minio_server.hostname},IP:#{minio_server.vm.ephemeral_net4}", "keyUsage=digitalSignature,keyEncipherment", "subjectKeyIdentifier=hash", "extendedKeyUsage=serverAuth"],
```

Comment:
I know I suggested it, but I'm not fully confident about adding IP to SAN as a security best practice on a potentially multi-tenant system. Also it could create problems after failovers.

How about doing it only for dev environment?

Reply:
updated

Reply:
Did just using DNS not work? We thought it might work since we use /etc/hosts.